install_cluster.py

```diff
@@ -0,0 +1,262 @@
+from os.path import expanduser, exists, join
+
+import yaml
+
+from common import (CfyNode, copy, get_dict_from_yaml,
+                    JUMP_HOST_CONFIG_PATH, JUMP_HOST_DIR,
+                    JUMP_HOST_LICENSE_PATH, JUMP_HOST_SSH_KEY_PATH,
+                    logger, move, run, sudo)
+
+INSTANCES_TYPES = ['postgresql', 'rabbitmq', 'manager']  # Order is important
+CERT_PATH = '{0}/.cloudify-test-ca'.format(expanduser('~'))
+RPM_NAME = 'cloudify-manager-install.rpm'
+CONFIG_FILES = 'config_files'
+
+JUMP_HOST_RPM_PATH = join(JUMP_HOST_DIR, RPM_NAME)
+LOCAL_INSTALL_CLUSTER_DIR = join(JUMP_HOST_DIR, 'cluster_install')
+LOCAL_CERTS_DIR = join(LOCAL_INSTALL_CLUSTER_DIR, 'certs')
+LOCAL_CONFIG_DIR = join(LOCAL_INSTALL_CLUSTER_DIR, CONFIG_FILES)
+
+REMOTE_INSTALL_CLUSTER_DIR = join('/tmp', 'cluster_install')
+REMOTE_CERTS_DIR = join(REMOTE_INSTALL_CLUSTER_DIR, 'certs')
+REMOTE_RPM_PATH = join(REMOTE_INSTALL_CLUSTER_DIR, RPM_NAME)
+REMOTE_CONFIG_DIR = join(REMOTE_INSTALL_CLUSTER_DIR, CONFIG_FILES)
+
+
+def _generate_instance_certificate(instance):
+    run(['cfy_manager', 'generate-test-cert', '-s',
+         instance.private_ip+','+instance.public_ip])
+    move(join(CERT_PATH, instance.private_ip+'.crt'),
+         join(CERT_PATH, instance.name+'_cert.pem'))
+    new_key_path = join(CERT_PATH, instance.name+'_key.pem')
+    move(join(CERT_PATH, instance.private_ip+'.key'), new_key_path)
+    sudo(['chmod', '444', new_key_path])
+
+
+def _generate_certs(instances_dict):
+    logger.info('Generating certificates')
+    for instances_list in instances_dict.values():
+        for instance in instances_list:
+            _generate_instance_certificate(instance)
+    copy(join(CERT_PATH, 'ca.crt'), join(CERT_PATH, 'ca.pem'))
+    os.mkdir(LOCAL_CERTS_DIR)
+    copy(CERT_PATH+'/.', LOCAL_CERTS_DIR)
+    shutil.rmtree(CERT_PATH)
+
+
+def _write_certs_to_config(config_dict, config_section, node_name):
+    cert_path = join(REMOTE_CERTS_DIR, node_name+'_cert.pem')
+    key_path = join(REMOTE_CERTS_DIR, node_name+'_key.pem')
+    ca_path = join(REMOTE_CERTS_DIR, 'ca.pem')
+
+    config_dict['prometheus']['cert_path'] = cert_path
+    config_dict['prometheus']['key_path'] = key_path
+    config_dict['prometheus']['ca_path'] = ca_path
+
+    if config_section == 'manager':
+        ssl_inputs = {}
+        cert_names = ['internal_cert_path', 'external_cert_path',
+                      'postgresql_client_cert_path']
+        key_names = ['internal_key_path', 'external_key_path',
+                     'postgresql_client_key_path']
+        ca_names = ['ca_cert_path', 'external_ca_cert_path']
+        for cert_name in cert_names:
+            ssl_inputs[cert_name] = cert_path
+        for key_name in key_names:
+            ssl_inputs[key_name] = key_path
+        for ca_name in ca_names:
+            ssl_inputs[ca_name] = ca_path
+        config_dict['ssl_inputs'] = ssl_inputs
+
+        config_dict['prometheus']['blackbox_exporter']['ca_cert_path'] = \
```

Comment:
Doesn't prometheus.ca_path, prometheus.cert_path, and prometheus.key_path need populating too, or does it fall back to ssl_inputs?
I think blackbox_exporter might fall back too?

Reply:
Yep, blackbox_exporter falk back is the internal CA, I'll drop it from the config. And regarding prometheus, they generate new ones. https://github.com/cloudify-cosmo/cloudify-manager-install/blob/1317d1af2e831b02e186bf7e0bfbb6c1e5fa8398/cfy_manager/components/prometheus/prometheus.py#L267